src/polymer/polymer-core-feature-scanner.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as estree from 'estree';
+
+import {Visitor} from '../javascript/estree-visitor';
+import * as esutil from '../javascript/esutil';
+import {JavaScriptDocument} from '../javascript/javascript-document';
+import {JavaScriptScanner} from '../javascript/javascript-scanner';
+import {Severity} from '../model/model';
+
+import {toScannedPolymerProperty} from './js-utils';
+import {ScannedPolymerCoreFeature} from './polymer-core-feature';
+
+/**
+ * Scans for Polymer 1.x core "features".
+ *
+ * In the Polymer 1.X core library, the `Polymer.Base` prototype is dynamically
+ * augmented with properties via calls to `Polymer.Base._addFeature`. These
+ * calls are spread across multiple files and split between the micro, mini,
+ * and standard "feature layers". Polymer 2.x does not use this pattern.
+ *
+ * Example: https://github.com/Polymer/polymer/blob/1.x/src/mini/debouncer.html
+ */
+export class PolymerCoreFeatureScanner implements JavaScriptScanner {
+  async scan(
+      document: JavaScriptDocument,
+      visit: (visitor: Visitor) => Promise<void>) {
+    const features: ScannedPolymerCoreFeature[] = [];
+
+    await visit({
+      enterCallExpression(node: estree.CallExpression, parent: estree.Node) {
+        const isAddFeatureCall = esutil.matchesCallExpression(
+            <estree.MemberExpression>node.callee,
+            ['Polymer', 'Base', '_addFeature']);
+        if (!isAddFeatureCall) {
+          return;
+        }
+
+        const description = esutil.getAttachedComment(parent) || '';
+        const feature = new ScannedPolymerCoreFeature(
+            document.sourceRangeForNode(node),
+            node,  // TODO(aomarks) Correct? The astNode param is not typed.
+            description,
+            // TODO(aomarks) Should I set jsdoc?
```

Comment:
What's description supposed to represent vs jsdoc? The documentation for both of these fields just says that they each probably shouldn't be there :P.